aws/iam-instance-profile.yml

```diff
@@ -0,0 +1,24 @@
+---
+- type: remove
+  path: /resource_pools/name=vms/cloud_properties/access_key_id?
+
+- type: remove
+  path: /resource_pools/name=vms/cloud_properties/secret_access_key?
+
+- type: replace
+  path: /resource_pools/name=vms/cloud_properties/iam_instance_profile?
+  value: ((iam_instance_profile))
+
+- type: remove
+  path: /instance_groups/name=bosh/properties/aws/access_key_id
+
+- type: remove
+  path: /instance_groups/name=bosh/properties/aws/secret_access_key
+
+- type: replace
+  path: /instance_groups/name=bosh/properties/aws/credentials_source?
+  value: env_or_profile
+
+- type: replace
+  path: /instance_groups/name=bosh/properties/aws/default_iam_instance_profile?
```

Comment:
@cppforlife @genevievelesperance so if we only wanted to give this to the director VM, then this specifically, we should not be doing, right? or is there more?